pkg/api/handlers/compat/containers_stats.go

```diff
@@ -80,6 +80,15 @@ func StatsContainer(w http.ResponseWriter, r *http.Request) {
 			ThrottlingData: docker.ThrottlingData{},
 		}
 	}
+	onlineCPUs := 0
+	if ctrPID, err := ctnr.PID(); err != nil {
+		utils.InternalServerError(w, fmt.Errorf("failed to obtain Container %s PID: %w", name, err))
+	} else {
+		onlineCPUs, err = libpod.GetOnlineCPUs(uint64(ctrPID))
+		if err != nil {
+			utils.InternalServerError(w, fmt.Errorf("failed to obtain Container %s online cpus: %w", name, err))
+		}
+	}
```

Comment:
Would it make sense to move this logic into GetOnlineCPUs in stats_common.go and then call that down below?  Or perhaps change this to a function in container_stats.go and call it from below.  It just seems like a lot of processing goes on from the time that onlineCPUs is set to when it's used.

Reply:
sure no problem.